### PR TITLE
[Remove Blocking usages]LPS-192257 portal-kernel |FullNameGeneratorFactory

### PR DESCRIPTION
**Pull request by JanisZhang**
FullNameGenerator here does not need to be blocking as there is always a default one populated by spring xml.

https://liferay.atlassian.net/browse/LPS-192257

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:test:startup

File: portal-kernel/src/com/liferay/portal/kernel/security/auth/FullNameGeneratorFactory.java [hunk 1]
```diff
@@ -27,6 +27,6 @@ private FullNameGeneratorFactory() {
 	private static volatile FullNameGenerator _fullNameGenerator =
 		ServiceProxyFactory.newServiceTrackedInstance(
 			FullNameGenerator.class, FullNameGeneratorFactory.class,
-			"_fullNameGenerator", true);
+			"_fullNameGenerator", false);
```
Comment (dantewang) [inline portal-kernel/src/com/liferay/portal/kernel/security/auth/FullNameGeneratorFactory.java:30]: @JanisZhang If the `FullNameGenerator` is instantiated by Spring, could you just change this inject the instance by Spring, and rename the class to `FullNameGeneratorUtil`?

Comment (JanisZhang) [reply portal-kernel/src/com/liferay/portal/kernel/security/auth/FullNameGeneratorFactory.java:30]: Hi @dantewang, `FullNameGenerator` has two subclasses named `DefaultFullNameGenerator` and `FamilyNameFirstFullNameGenerator`. the `DefaultFullNameGenerator` is the only one injected using Spring configuration. I am not sure whether the FamilyNameFirstFullNameGenerator will be invoked if we change it in this way.

Comment (dantewang) [reply portal-kernel/src/com/liferay/portal/kernel/security/auth/FullNameGeneratorFactory.java:30]: @JanisZhang Which one is actually in that field, the `Default` or the `FamilyNameFirst` one?

Comment (JanisZhang) [reply portal-kernel/src/com/liferay/portal/kernel/security/auth/FullNameGeneratorFactory.java:30]: @dantewang  I have done some tests, the actual value in the field is the default one.

**Comment by dantewang** [comment]
@JanisZhang Closing for now, I asked a question in help-security.

**Comment by JanisZhang** [comment]
force-pushed. I only changed the commit message, it shouldn't affect the test results. Thanks.

**Comment by Kvale1733** [comment]
ci:test:security

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by JanisZhang** [comment]
force-pushed, use null as dummy service. thanks!

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:startup

**Comment by JanisZhang** [comment]
ci:test:security

**Comment by dantewang** [comment]
@JanisZhang Please mark it "no better way" for now, as we want to apply "non-blocking with null as proxy" at the very end of this task.

**Pull request closed without merging**